github4s/src/main/scala/github4s/http/HttpClient.scala

```diff
@@ -0,0 +1,148 @@
+      headers: Map[String, String] = Map.empty,
+      params: Map[String, String] = Map.empty,
+      pagination: Option[Pagination] = None
+  ): F[GHResponse[Res]] =
+    run[Unit, Res](
+      RequestBuilder(url = buildURL(method))
+        .withAuth(accessToken)
+        .withHeaders(headers)
+        .withParams(params ++ pagination.fold(Map.empty[String, String])(p =>
+          Map("page" -> p.page.toString, "per_page" -> p.per_page.toString))))
+
+  def patch[Req: Encoder, Res: Decoder](
+      accessToken: Option[String] = None,
+      method: String,
+      headers: Map[String, String] = Map.empty,
+      data: Req): F[GHResponse[Res]] =
+    run[Req, Res](
+      RequestBuilder(buildURL(method)).patchMethod
+        .withAuth(accessToken)
+        .withHeaders(headers)
+        .withData(data))
+
+  def put[Req: Encoder, Res: Decoder](
+      accessToken: Option[String] = None,
+      url: String,
+      headers: Map[String, String] = Map(),
+      data: Req): F[GHResponse[Res]] =
+    run[Req, Res](
+      RequestBuilder(buildURL(url)).putMethod
+        .withAuth(accessToken)
+        .withHeaders(headers)
+        .withData(data))
+
+  def post[Req: Encoder, Res: Decoder](
+      accessToken: Option[String] = None,
+      url: String,
+      headers: Map[String, String] = Map.empty,
+      data: Req
+  ): F[GHResponse[Res]] =
+    run[Req, Res](
+      RequestBuilder(buildURL(url)).postMethod
+        .withAuth(accessToken)
+        .withHeaders(headers)
+        .withData(data))
+
+  def postAuth[Req: Encoder, Res: Decoder](
+      method: String,
+      headers: Map[String, String] = Map.empty,
+      data: Req
+  ): F[GHResponse[Res]] =
+    run[Req, Res](RequestBuilder(buildURL(method)).postMethod.withHeaders(headers).withData(data))
+
+  def postOAuth[Req: Encoder, Res: Decoder](
+      url: String,
+      headers: Map[String, String] = Map.empty,
+      data: Req
+  ): F[GHResponse[Res]] =
+    run[Req, Res](
+      RequestBuilder(url).postMethod
+        .withHeaders(Map("Accept" -> "application/json") ++ headers)
+        .withData(data))
+
+  def delete(
+      accessToken: Option[String] = None,
+      method: String,
+      headers: Map[String, String] = Map.empty): F[GHResponse[Unit]] =
+    run[Unit, Unit](
+      RequestBuilder(buildURL(method)).deleteMethod.withHeaders(headers).withAuth(accessToken))
+
+  def deleteWithResponse[Res: Decoder](
+      accessToken: Option[String] = None,
+      url: String,
+      headers: Map[String, String] = Map.empty
+  ): F[GHResponse[Res]] =
+    run[Unit, Res](
+      RequestBuilder(buildURL(url)).deleteMethod
+        .withAuth(accessToken)
+        .withHeaders(headers))
+
+  val defaultPagination   = Pagination(1, 1000)
+  val defaultPage: Int    = 1
+  val defaultPerPage: Int = 30
+  val resource: Resource[F, Client[F]] = BlazeClientBuilder[F](ec)
+    .withConnectTimeout(connTimeout)
+    .resource
+
+  private def buildURL(method: String) = urls.baseUrl + method
```

Comment:
It's a simple and private function but just to continue with the same signature as the others, maybe return type can be added to function definition